src/main/java/com/example/bank/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,70 @@
+package com.example.bank.exception;
+
+import jakarta.annotation.Nullable;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.util.WebUtils;
+
+import java.util.Collections;
+import java.util.List;
+
+@RestControllerAdvice
+/** 1) Nie widzę różnicy między @RestControllerAdvice i @ControllerAdvice.
+ * Teoretycznie teraz powinienem dostać odpowiedź JSON lub XLM, ale nie zauważyłem różnicy w odpowiedzi z serwera.
+ *
+ * 2) Swoją drogą, skoro zwracana wartość ma być web response, to czy powinienem wrapować poniższe metody
+ *   w ResponseEntity<> ?? W controllerze też ustawiam statusy.
+ *
+ * 3) A po trzecie, czy ktoś te statusy na froncie w ogóle czyta?  </>*/
+public class GlobalExceptionHandler {
+    @ExceptionHandler(
+            {ClientNotFoundException.class, IllegalArgumentException.class}
+    )
+
+    final ResponseEntity<List<String>> handleException(Exception ex, WebRequest request) {
+        HttpHeaders headers = new HttpHeaders();
+        String exceptionTypeString = ex.getClass().getSimpleName();
+        switch (exceptionTypeString) {
```

Comment:
tak bym tego nie robil, raczej bym zrobil switcha ktory sprawdza co to za klasa a nie co to za nazwa klasy, tak jest bardziej profesjonalnie

Reply:
Zrobiłem inaczej, bo switch nie przyjmuje klasy dziedziczącej po Exception.